### PR TITLE
feat(#247): show availabilities in a calendar

### LINKED ISSUE #247: When a room is selected, show different availabilities
### User story

When a room is selected, show different availabilities

### Narrative

When I select a room to make a booking request, I should see the current availabilities (time restriction per day/booking period restriction/current bookings/blackout period).

### Out of scope

create or update those availabilities

### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am a user|I make a booking request|I should see current room time restrictions in a table|
|2|I am a user|I make a booking request|I should see the current time period when a room is available/unavailable inside a calendar view|

### Mockups

image

### PR DESCRIPTION
**Pull request by alecadub**
##### **_IMPORTANT: Always create an issue before a PR and provide enough information so that others can review_**

Show availabilities/non-availabilities/blackout/bookings in a mini calendar

image

##### _Demonstrate the code is solid. </br> Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI._

- See checks

##### _List all issues this PR resolves_
- closes #247 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [x] System Tests
- [x] Documentation

### CODE REVIEW

**Comment by EvanDime** [comment]
The implementation seems fine for now since it shows that at least the day is open and has a blackout period. 
Later on, to improve the tool, I want to add a feature that when you click on a day, it makes an API call to get the times that are available that day(not just when it is open but show what times are blocked by other reservations).
For now I am at work and can't review this, but I like the screenshot so GREAT job Alec

**Comment by alecadub** [comment]
> The implementation seems fine for now since it shows that at least the day is open and has a blackout period. 
> 
> Later on, to improve the tool, I want to add a feature that when you click on a day, it makes an API call to get the times that are available that day(not just when it is open but show what times are blocked by other reservations).
> 
> For now I am at work and can't review this, but I like the screenshot so GREAT job Alec

Yes we can definitely improve in the future and add many many features to it, but this is the first part :)

File: package.json [hunk 1]
```diff
@@ -78,15 +78,19 @@
...
-          }
-          ]
+            [
+                "@babel/plugin-transform-runtime",
+                {
+                    "corejs": false,
+                    "regenerator": true
+                }
+            ]
         ]
```
Comment (jonathanp16) [inline package.json:88]: Might cause merge conflict

File: resources/js/Components/AvailabilitiesModal.vue [hunk 2]
```diff
@@ -0,0 +1,288 @@
+<template>
+  <jet-dialog-modal :show="room" @close="closeModal" max-width="">
```
Comment (jonathanp16) [inline resources/js/Components/AvailabilitiesModal.vue:2]: max-width not needed

File: resources/js/Components/AvailabilitiesModal.vue [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    JetInputError,
+    JetLabel,
+    JetDropdown,
+    JetDropdownLink,
+    JetNavLink,
+    JetDialogModal,
+    JetSecondaryButton,
+    Calendar,
+    DatePicker,
+    moment
```
Comment (jonathanp16) [inline resources/js/Components/AvailabilitiesModal.vue:117]: Not sure if you need to declare moment as a component

**Review by jonathanp16 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (alexstojda) [reply package.json:88]: As long as there are no potential merge conflicts reported by github we can leave this :)

File: resources/js/Components/AvailabilitiesModal.vue (hunk 2 above)
Comment (alexstojda) [reply resources/js/Components/AvailabilitiesModal.vue:2]:
```suggestion
  <jet-dialog-modal :show="room" @close="closeModal">
```

File: resources/js/Components/AvailabilitiesModal.vue (hunk 3 above)
Comment (alexstojda) [reply resources/js/Components/AvailabilitiesModal.vue:117]:
```suggestion
```

**Review by alexstojda — APPROVED** [review]
Looks good! Well done

**Review by jonathanp16 — APPROVED** [review]

**Pull request merged**